### DISCUSSION TITLE
How to reset callback URI for default sign in page after using a custom sign in?

### DISCUSSION THREAD

**Discussion by theklinks22** [nextauthjs/next-auth#8569] [Help]
### Question 💬

Hi all I am having an issue registering new users with email provider. For signIn I am using the provided sign in screen from AuthJs. For email users, if they attempt to sign in, but no account is registered they are redirected to a registration form.

For e-mail registration, I want a User to Specify their name before an account is created, and on their first sign in, get redirected to the getting-started page that will have valuable info on using the application. In subsequent log-ins, they should be redirected to the root page.

With OAuth providers the newUser feature works as expected, but I was running into issues with the email provider. If we don't detect a user it sends us to a registration page where the user can fill out account details. On a successful submission of the form they are prompted to check their e-mail for the magic link.

Because this user has now been added to the database, AuthJs doesn't recognize them as a new user, even when the updateUser event is fired. To get around this I added the signIn function and added the callbackUrl field to it on my registration form. 

The problem I am having now is that all subsequent signIn and signOut requests all redirect to the getting started page now when a user registers and it stays that way until I restart the server. Do I now have to roll my own signIn page for all providers?

Portion of Form Submission where we call signIn
```JS
setSubmitting(false)

    if (status === 200) {
      setSubmitted(true)
      await signIn('email', {
        email: email,
        redirect: false,
        callbackUrl: '/account/getting-started',
      })
    } else {
      setErrorState((prev) => ({ ...prev, submitError: message }))
    }
```

My NextAuthOptions: I added console output on createUser, updateUser, signIn, and redirect to monitor what was happening
```JS
export const authOptions: NextAuthOptions = {
  pages: {
    newUser: '/account/getting-started',
  },
  adapter: PrismaAdapter(prisma),
  theme: {
    colorScheme: 'light',
    brandColor: '#075985',
    logo: logo.src,
  },
  providers: [
    EmailProvider({
      server: {
        host: process.env.EMAIL_SERVER_HOST,
        port: process.env.EMAIL_SERVER_PORT,
        auth: {
          user: process.env.EMAIL_SERVER_USER,
          pass: process.env.EMAIL_SERVER_PASSWORD,
        },
      },
      from: process.env.EMAIL_FROM,
    }),

    GoogleProvider({
      clientId: process.env.GOOGLE_CLIENT_ID!,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET!,
    }),
    FacebookProvider({
      clientId: process.env.FACEBOOK_CLIENT_ID!,
      clientSecret: process.env.FACEBOOK_CLIENT_SECRET!,
    }),
    // AppleProvider({
    //   clientId: process.env.APPLE_CLIENT_ID!,
    //   clientSecret: process.env.APPLE_CLIENT_SECRET!,
    // }),
  ],
  events: {
    async createUser(message) {
      console.log('created User:\n', message)
    },
    async updateUser(message) {
      console.log('updated User:\n', message)
    },
    async signIn(user) {
      console.log('signIn event:', user)
    },
  },
  callbacks: {
    async redirect({ url, baseUrl }) {
      console.log('redirect: ' + url + '\nbase: ' + baseUrl)
      // Allows relative callback URLs
      if (url.startsWith('/')) return `${baseUrl}${url}`
      // // Allows callback URLs on the same origin
      else if (new URL(url).origin === baseUrl) return url
      return baseUrl
    },
    async signIn({ user, email }) {
      if (!email) {
        return true
      }
      const userExists = await prisma.user.findUnique({
        where: { email: user.email! },
      })
      if (userExists) {
        return true
      } else {
        return '/account/register'
      }
    },
  },
}

const handler = NextAuth(authOptions)
export { handler as GET, handler as POST }
```

```
System:
    OS: Linux 5.19 Linux Mint 21.1 (Vera)
    CPU: (16) x64 Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz
    Memory: 13.37 GB / 31.26 GB
    Container: Yes
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    npm: 9.6.7 - ~/.nvm/versions/node/v18.16.0/bin/npm
  Browsers:
    Chrome: 116.0.5845.96
```

### How to reproduce ☕️

1. Use default sign In page
2. You will be redirected to root uri
3. Call signIn from elsewhere on your application and set callbackUri using e-mail provider
4. Sign In from default provider again (You will be redirected to the callbackUri)

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by goerlitz** [comment]
@theklinks22 , did you find any solution? I'm facing a very similar issue.
